contrib/devtools/lint.dockerfile

```diff
@@ -0,0 +1,49 @@
+# DOCKER_BUILDKIT=1 docker build --pull --no-cache -t linter - < lint.dockerfile
+#	Use --build-arg PYTHON=3.11.0 etc to change installed versions
+# docker run --rm -v $(pwd):/bitcoin -w /bitcoin -it linter
+FROM ubuntu:jammy
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV LC_ALL=C.UTF-8
+
+ARG CODESPELL=2.2.1
+ARG FLAKE=5.0.4
+ARG MYPY=0.971
+ARG PYTHON=3.6.15
+ARG PYZMQ=24.0.1
+ARG SHELLCHECK=v0.8.0
+ARG VULTURE=2.6
```

Comment:
You claim that

> live in two separate places is a recipe for version mismatches.

But in this pull you are duplicating the versions to two places in this repo. (As well as all other stuff)